include/xtensor-signal/find_peaks.hpp

```diff
@@ -0,0 +1,725 @@
+                    throw std::runtime_error(
+                        "arrays in prominence_data must have the same shape as peaks");
+                }
+
+                auto widths = xt::empty<double>({ peaks.shape(0) });
+                auto width_heights = xt::empty<double>({ peaks.shape(0) });
+                auto left_ips = xt::empty<double>({ peaks.shape(0) });
+                auto right_ips = xt::empty<double>({ peaks.shape(0) });
+
+                for (size_t p = 0; p < peaks.shape(0); p++)
+                {
+                    size_t i_min = left_bases(p);
+                    size_t i_max = right_bases(p);
+                    auto peak = peaks(p);
+
+                    //validate the bounds and order
+                    if (i_min < 0 || peak < i_min || i_max < peak || i_max >= x.shape(0))
+                    {
+                        throw std::runtime_error(
+                            "Invalid prominence data is invalid for peak");
+                    }
+
+                    auto height = x(peak) - prominences(p) * rel_height;
+                    width_heights(p) = height;
+                    auto i = peak;
+                    //find intersecption point on left side
+                    while (i_min < i && height < x(i))
+                    {
+                        i--;
+                    }
+
+                    double left_ip = static_cast<double>(i);
+                    if (x(i) < height)
+                    {
+                        left_ip += (height - x(i)) / (x(i + 1) - x(i));
+                    }
+
+                    i = peak;
+                    while (i < i_max && height < x(i))
+                    {
+                        i++;
+                    }
+
+                    double right_ip = static_cast<double>(i);
+                    if (x(i) < height)
+                    {
+                        right_ip -= (height - x(i)) / (x(i - 1) - x(i));
+                    }
+
+                    widths(p) = right_ip - left_ip;
+                    left_ips(p) = left_ip;
+                    right_ips(p) = right_ip;
+                }
+                return std::make_tuple(widths, width_heights, left_ips, right_ips);
+            }
+
+            /**
+             * @brief Find local maxima in a 1D array.
+             * @details This function finds all local maxima in a 1D array and returns the indices
+             *     for their edges and midpoints (rounded down for even plateau sizes).
+             * @param x The array to search for local maxima.
+             * @returns Indices of midpoints of local maxima in `x`.
+             * left_edges Indices of edges to the left of local maxima in `x`.
+             * right_edges Indices of edges to the right of local maxima in `x`.
+             * @note Derived from https://github.com/scipy/scipy/blob/main/scipy/signal/_peak_finding_utils.pyx
+             */
+            template<typename T>
+            auto local_maxima_1d(T&& x)
```

Comment:
The return type can be explicited here (using the trailing return syntax is an option too)